internal/services/service.go

```diff
@@ -34,49 +35,47 @@ type Service struct {
 // Start will execute a binary and wait for its completion or until the context
 // is canceled
 func (s Service) Start(ctx context.Context) error {
-	cmd := exec.Command(s.binaryName)
+	cmd := exec.CommandContext(ctx, s.binaryName)
 	cmd.Stderr = os.Stderr
 	cmd.Stdout = os.Stdout
-
-	if err := cmd.Start(); err != nil {
-		return err
-	}
-
-	go func() {
-		<-ctx.Done()
-		logger.Debug.Printf("%v: %v\n", s.String(), ctx.Err())
-		if err := cmd.Process.Signal(syscall.SIGTERM); err != nil {
+	cmd.Cancel = func() error {
+		err := cmd.Process.Signal(syscall.SIGTERM)
+		if err != nil {
 			msg := "failed to send SIGTERM to %v: %v\n"
 			logger.Warning.Printf(msg, s.name, err)
 		}
-	}()
-
-	err := cmd.Wait()
-	exitCode := cmd.ProcessState.ExitCode()
-	signal := cmd.ProcessState.Sys().(syscall.WaitStatus).Signal()
-	if err != nil && exitCode != 0 && signal != syscall.SIGTERM {
 		return err
 	}
+	err := cmd.Run()
+	if err != nil {
+		exitCode := cmd.ProcessState.ExitCode()
+		signal := cmd.ProcessState.Sys().(syscall.WaitStatus).Signal()
+		if exitCode != 0 && signal != syscall.SIGTERM {
+			// only return error if the service exits for reason other than shutdown
+			return err
+		}
```

Comment:
I changed this to make the code clearer, but the previous version was fine as well.